src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        for (int i = 0; i < machines.size(); i++) {
+            machines.set(i, value);
+        }
+    }
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = new ArrayList<>();
+        MachineProducer<? extends Machine> producer;
+
+        if (type == Bulldozer.class) {
+            producer = new BulldozerProducer();
+            machines = producer.get();
+        }
+        if (type == Excavator.class) {
+            producer = new ExcavatorProducer();
+            machines = producer.get();
+        }
+        if (type == Truck.class) {
+            producer = new TruckProducer();
+            machines = producer.get();
+        }
+        return new ArrayList<>(machines);
```

Comment:
Why do you need the machines and producer variables when you can return them all at once in an if block?